Mathlib/CategoryTheory/Galois/Prorepresentability.lean

```diff
@@ -0,0 +1,235 @@
+  f.comp
+
+@[simp]
+lemma id_val (A : PointedGaloisObject F) : 𝟙 A = 𝟙 A.obj :=
+  rfl
+
+@[simp]
+lemma comp_val {A B C : PointedGaloisObject F} (f : A ⟶ B) (g : B ⟶ C) :
+    (f ≫ g).val = f.val ≫ g.val :=
+  rfl
+
+variable (F)
+
+/-- The category of pointed Galois objects is cofiltered. -/
+instance : IsCofilteredOrEmpty (PointedGaloisObject F) where
+  cone_objs := fun ⟨A, a, _⟩ ⟨B, b, _⟩ ↦ by
+    obtain ⟨Z, f, z, hgal, hfz⟩ := exists_hom_from_galois_of_fiber F (A ⨯ B)
+      <| (fiberBinaryProductEquiv F A B).symm (a, b)
+    refine ⟨⟨Z, z, hgal⟩, ⟨f ≫ prod.fst, ?_⟩, ⟨f ≫ prod.snd, ?_⟩, trivial⟩
+    · simp only [F.map_comp, hfz, FintypeCat.comp_apply, fiberBinaryProductEquiv_symm_fst_apply]
+    · simp only [F.map_comp, hfz, FintypeCat.comp_apply, fiberBinaryProductEquiv_symm_snd_apply]
+  cone_maps := fun ⟨A, a, _⟩ ⟨B, b, _⟩ ⟨f, hf⟩ ⟨g, hg⟩ ↦ by
+    obtain ⟨Z, h, z, hgal, hhz⟩ := exists_hom_from_galois_of_fiber F A a
+    refine ⟨⟨Z, z, hgal⟩, ⟨h, hhz⟩, Hom.ext ?_⟩
+    apply evaluationInjective_of_isConnected F Z B z
+    show F.map (h ≫ f) z = F.map (h ≫ g) z
+    simp only [map_comp, FintypeCat.comp_apply, hhz, hf, hg]
+
+/-- The canonical functor from pointed Galois objects to `C`. -/
+def incl : PointedGaloisObject F ⥤ C where
+  obj := fun A ↦ A
+  map := fun ⟨f, _⟩ ↦ f
+
+@[simp]
+lemma incl_obj (A : PointedGaloisObject F) : (incl F).obj A = A :=
+  rfl
+
+@[simp]
+lemma incl_map {A B : PointedGaloisObject F} (f : A ⟶ B) : ((incl F).map f) = f.val :=
+  rfl
+
+/-- `F ⋙ FintypeCat.incl` as a cocone over `(can F).op ⋙ coyoneda`.
+This is a colimit cocone (see `PreGaloisCategory.isColimìt`) -/
+def cocone : Cocone ((incl F).op ⋙ coyoneda) where
+  pt := F ⋙ FintypeCat.incl
+  ι := {
+    app := fun ⟨A, a, _⟩ ↦ { app := fun X (f : (A : C) ⟶ X) ↦ F.map f a }
+    naturality := fun ⟨A, a, _⟩ ⟨B, b, _⟩ ⟨f, (hf : F.map f b = a)⟩ ↦ by
+      ext Y (g : (A : C) ⟶ Y)
+      suffices h : F.map g (F.map f b) = F.map g a by
+        simpa
+      rw [hf]
+  }
+
+@[simp]
+lemma cocone_app (A : PointedGaloisObject F) (B : C) (f : (A : C) ⟶ B) :
+    ((cocone F).ι.app ⟨A⟩).app B f = F.map f A.pt :=
+  rfl
+
+/-- `cocone F` is a colimit cocone, i.e. `F` is pro-represented by `incl F`. -/
+noncomputable def isColimit : IsColimit (cocone F) := by
+  refine evaluationJointlyReflectsColimits _ (fun X ↦ ?_)
+  refine Types.FilteredColimit.isColimitOf _ _ ?_ ?_
+  · intro (x : F.obj X)
+    obtain ⟨Y, i, y, h1, _, _⟩ := fiber_in_connected_component F X x
+    obtain ⟨Z, f, z, hgal, hfz⟩ := exists_hom_from_galois_of_fiber F Y y
+    refine ⟨⟨Z, z, hgal⟩, f ≫ i, ?_⟩
+    simp only [mapCocone_ι_app, evaluation_obj_map, cocone_app, map_comp,
+      ← h1, FintypeCat.comp_apply, hfz]
+  · intro ⟨A, a, _⟩ ⟨B, b, _⟩ (u : (A : C) ⟶ X) (v : (B : C) ⟶ X) (h : F.map u a = F.map v b)
+    obtain ⟨⟨Z, z, _⟩, ⟨f, hf⟩, ⟨g, hg⟩, _⟩ :=
+      @IsFilteredOrEmpty.cocone_objs (PointedGaloisObject F)ᵒᵖ _ _
+      ⟨{ obj := A, pt := a}⟩ ⟨{obj := B, pt := b}⟩
```

Comment:
This syntax allows to avoid `@` in many situations:
```suggestion
      IsFilteredOrEmpty.cocone_objs (C := (PointedGaloisObject F)ᵒᵖ)
        ⟨{ obj := A, pt := a}⟩ ⟨{obj := B, pt := b}⟩
```